supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/statuspolling/JesPollingActor.scala

```diff
@@ -3,88 +3,65 @@ package cromwell.backend.impl.jes.statuspolling
 import akka.actor.{Actor, ActorLogging, ActorRef, Props}
 import cats.data.NonEmptyList
 import com.google.api.client.googleapis.batch.BatchRequest
-import com.google.api.client.googleapis.batch.json.JsonBatchCallback
 import com.google.api.client.googleapis.json.GoogleJsonError
-import com.google.api.client.http.HttpHeaders
-import com.google.api.services.genomics.model.Operation
-import cromwell.backend.impl.jes.{JesAttributes, Run}
-import cromwell.backend.impl.jes.statuspolling.JesApiQueryManager.{JesPollingWorkBatch, JesStatusPollQuery, NoWorkToDo}
+import com.google.api.services.genomics.Genomics
+import cromwell.backend.impl.jes.statuspolling.JesApiQueryManager._
 import cromwell.backend.impl.jes.statuspolling.JesPollingActor._
 import cromwell.core.Dispatcher.BackendDispatcher
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.numeric._
 
-import scala.collection.JavaConversions._
-import scala.concurrent.{ExecutionContext, Future, Promise}
+import scala.concurrent.{ExecutionContext, Future}
 import scala.util.{Failure, Success, Try}
 import scala.concurrent.duration._
+import scala.collection.JavaConversions._
 
 /**
-  * Polls JES for status. Pipes the results back (so expect either a RunStatus or a akka.actor.Status.Failure).
+  * Sends batched requests to JES as a worker to the JesApiQueryManager
   */
-class JesPollingActor(val pollingManager: ActorRef, val qps: Int) extends Actor with ActorLogging {
+class JesPollingActor(val pollingManager: ActorRef, val qps: Int Refined Positive) extends Actor with ActorLogging
+  with StatusPolling with RunCreation {
   // The interval to delay between submitting each batch
-  lazy val batchInterval = determineBatchInterval(determineEffectiveQps(qps))
+  lazy val batchInterval = determineBatchInterval(qps)
   log.debug("JES batch polling interval is {}", batchInterval)
 
-  self ! NoWorkToDo // Starts the check-for-work cycle
+  self ! NoWorkToDo // Starts the check-for-work cycle when the actor is fully initialized.
 
   implicit val ec: ExecutionContext = context.dispatcher
 
   override def receive = {
 
     case JesPollingWorkBatch(workBatch) =>
       log.debug(s"Got a polling batch with ${workBatch.tail.size + 1} requests.")
-      val batchResultFutures = handleBatch(workBatch)
-      val overallFuture = Future.sequence(batchResultFutures.toList)
-      overallFuture.andThen(interstitialRecombobulation)
+      handleBatch(workBatch).andThen(interstitialRecombobulation)
       ()
     case NoWorkToDo =>
       scheduleCheckForWork()
   }
 
-  private def handleBatch(workBatch: NonEmptyList[JesStatusPollQuery]): NonEmptyList[Future[Try[Unit]]] = {
-
-    // Assume that the auth for the first element is also able to query the remaining Runs
-    val batch: BatchRequest = createBatch(workBatch.head.run)
+  private def handleBatch(workBatch: NonEmptyList[JesApiQuery]): Future[List[Try[Unit]]] = {
```

Comment:
Presumably less feasible now due to the inversion of the list & future, but since futures already contain a try could this be simplified?